### ISSUE TITLE
zimcheck should catch zimfiles with invalid mimetype list.

### ISSUE THREAD

**Issue by mgautierfr** [openzim/zim-tools#18]
Recent zim files created by zimwriterfs are broken (https://github.com/openzim/zimwriterfs/issues/48)
zimcheck should be extended to catch them.

**Comment by MiguelRocha** [comment]
@mgautierfr If I understand correctly there may be Articles in which https://github.com/openzim/libzim/blob/2739d7544a100289b4e954da2ac8d10760c78d0b/src/article.cpp#L137

Throws an exception. (meaning that there are invalid idx). Am I right?

**Comment by kelson42** [comment]
@mgautierfr @MiguelRocha Or should we move this to the libzim itself like suggested in https://github.com/openzim/libzim/issues/359?

**Comment by MiguelRocha** [comment]
@kelson42 yes it seems to make more sense to have an "utility" checking function that is called from zimcheck. Is this what you meant @mgautierfr ?

**Comment by mgautierfr** [comment]
> Or should we move this to the libzim itself like suggested in openzim/libzim#359?

Yes, this is an internal problem. Not a "content" problem. It would be better in libzim itself.

>  yes it seems to make more sense to have an "utility" checking function in libzim that is called from zimcheck. Is this what you meant @mgautierfr ?

Yes, the structural validity of the zim file should be check in libzim (using a public function).
The "content validity" of the zim should be handle elsewhere (zimcheck or other specific tool). `zimcheck` would call the public "validation" function of libzim.

**Comment by kelson42** [comment]
> > Or should we move this to the libzim itself like suggested in [openzim/libzim#359](https://github.com/openzim/libzim/issues/359)?
> 
> Yes, this is an internal problem. Not a "content" problem. It would be better in libzim itself.
> 
> > yes it seems to make more sense to have an "utility" checking function in libzim that is called from zimcheck. Is this what you meant @mgautierfr ?
> 
> Yes, the structural validity of the zim file should be check in libzim (using a public function).
> The "content validity" of the zim should be handle elsewhere (zimcheck or other specific tool). `zimcheck` would call the public "validation" function of libzim.

To be 100% clear on this proposal, I 100% support it and would like to see a first version of such a method implemented as soon as possible based on the needs express at zimcheck level.

**Comment by MiguelRocha** [comment]
@mgautierfr do you agree with my first comment?

@kelson42 should we create also an issue in libzim repo?

**Comment by kelson42** [comment]
openzim/libzim#359 is enough to start with I believe.

**Comment by mgautierfr** [comment]
> do you agree with my first comment?

Yes, we should throw an exception. And it would be an invalid file format or equivalent exception.
However, we will not change that in the current API. Next API will throw exception when something goes wrong instead of returning default values.
